server/sttp-mock-server/src/test/scala/sttp/tapir/server/mockserver/MockServerSpec.scala

```diff
@@ -537,6 +544,252 @@ class MockServerSpec extends AnyFlatSpec with Matchers {
+                    "httpRequest" : {
+                      "method": "PUT",
+                      "path": "/api/v1/order",
+                      "body" : {
+                        "json" : $sampleIn,
+                        "matchType": "STRICT",
+                        "type" : "JSON"
+                      },
+                      "headers": {
+                        "Content-Length": ["28"],
+                        "Content-Type": ["application/json"],
+                        "Accept-Encoding": ["gzip, deflate"]
+                      }
+                    },
+                    "httpResponse" : {
+                      "statusCode": 400,
+                      "headers": {
+                        "Content-Type": ["application/json"]
+                      },
+                      "body" : $sampleErrorOut
+                    },
+                    "times": {
+                      "remainingTimes": 0,
+                      "unlimited": true
+                    },
+                    "timeToLive": {
+                      "timeUnit": "DAYS",
+                      "timeToLive": 0,
+                      "unlimited": true
+                    }
+                  }
+                ]""".noSpaces
+        )
+      }
+    }
+
+    val expectations = mockServerClient
+      .whenInputMatches(orderEndpoint)((), sampleIn)
+      .thenError(sampleErrorOut, statusCode = StatusCode.BadRequest)
+
+    expectations should have size 1
+    println(expectations.head.httpRequest.path)
+    expectations.head shouldEqual Expectation(
+      id = "5d122dd6-ce49-4508-b045-a76d9887aec4",
+      priority = 0,
+      httpRequest = ExpectationRequestDefinition(
+        method = Method.PUT,
+        path = Uri.unsafeParse(
+          "/api/v1/order"
+        ), // NOTE: uri interpolator produces different case class instance having the same string representation =(
+        body = Some(
+          ExpectationBodyDefinition.JsonBodyDefinition(
+            json = sampleIn.asJsonObject,
+            matchType = ExpectationBodyDefinition.JsonMatchType.Strict
+          )
+        ),
+        headers = Some(
+          Map(
+            HeaderNames.ContentLength -> List("28"),
+            HeaderNames.ContentType -> List("application/json"),
+            HeaderNames.AcceptEncoding -> List("gzip, deflate")
+          )
+        )
+      ),
+      httpResponse = ExpectationResponseDefinition(
+        statusCode = StatusCode.BadRequest,
+        body = Some(ExpectationBodyDefinition.RawJson(sampleErrorOut.asJsonObject)),
+        headers = Some(
+          Map(
+            HeaderNames.ContentType -> List("application/json")
+          )
+        )
+      ),
+      times = ExpectationTimes(unlimited = true, remainingTimes = Some(0)),
+      timeToLive = ExpectationTimeToLive(unlimited = true, timeToLive = Some(0), timeUnit = Some("DAYS"))
+    )
+  }
+
+  it should "match json body with drop null" in {
+    import TapirJsonCirceWithDropNull._
```

Comment:
is this ever used?

Reply:
I wanted to make it explicit for these two cases that they are using different configurations related to this drop option.

Reply:
but are they? are you using the `jsonBody` method defined by these objects anywhere?